src/build/build_implicitconstraints.jl

```diff
@@ -120,29 +155,71 @@ function build_implicitconstraints!(m, jm)
+                /$(get_consumer_benchmark(demand_function.consumer))) # benchmark income (?)
+                *
+                ($(get_commodity_benchmark(demand.commodity))  # p__bar_i?
+                   /
+                   $(get_jump_variable_for_commodity(jm, demand.commodity))) #p_bar
+                        - 
+                    $(jm[get_final_demand_name(demand)])
+                    )
                 )
-            )
-        )
-        for demand in demand_function.demands
-            ex = :(
-                JuMP.@NLexpression(
-                    $(jm),
-                        $(demand.quantity) /
-                        $temp1 *
-                        $(get_jump_variable_for_consumer(jm, demand_function.consumer)) /
-                        $(get_jump_variable_for_commodity(jm, demand.commodity)) - 
-                        $(jm[get_final_demand_name(demand)])
+                exb = eval( swap_our_param_with_jump_param(jm, ex) )
+
+                Complementarity.add_complementarity(jm, jm[get_final_demand_name(demand)], exb, string("F_", get_final_demand_name(demand)))
+                push!(m._nlexpressions, exb)
+            end
+        elseif eval(swap_our_param_with_val(demand_function.elasticity))==0
+            for demand in demand_function.demands
+                Θ(d, df) = :( $(d.quantity) * $(get_commodity_benchmark(d.commodity)) /$(get_consumer_benchmark(df.consumer)))
+                ex = :(
+            JuMP.@NLexpression(
+                $(jm),
+                $(demand.quantity) * 
+                ($(get_jump_variable_for_consumer(jm, demand_function.consumer)) # (consumer's) income
+                /$(get_consumer_benchmark(demand_function.consumer))) # benchmark income (?)
+                *
+                (       +($( (:($(Θ(demand, demand_function))*$(get_commodity_benchmark(demand.commodity)) ) for demand in demand_function.demands)...))  # sum(demand theta * p__bar_j)?
+                   /
+                        +($( (:($(Θ(demand, demand_function))*$(get_jump_variable_for_commodity(jm, demand.commodity))) for demand in demand_function.demands)...) ))# sum(demand theta * p_bar
+                        - 
+                    $(jm[get_final_demand_name(demand)])
                 )
             )
+            exb = eval( swap_our_param_with_jump_param(jm, ex) )
 
+            Complementarity.add_complementarity(jm, jm[get_final_demand_name(demand)], exb, string("F_", get_final_demand_name(demand)))
+            push!(m._nlexpressions, exb)
+            end
+        else
+            for demand in demand_function.demands
+                ex = :(
+            JuMP.@NLexpression(
+                $(jm),
+                $(demand.quantity) * 
+                ($(get_jump_variable_for_consumer(jm, demand_function.consumer)) # (consumer's) income
+                /$(get_consumer_benchmark(demand_function.consumer))) # benchmark income (?)
```

Comment:
Generally safer to put the operator on the previous line. In this case it will work (I think) because of the parenthesis around it, but if they hadn't been there, then Julia would interpret the previous line as being finished and would start a new expression in the new line. If you end the previous line with the `/` operator, then it is clear that the next line _must_ be part of the previous expression.